File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/api/PhoneSupportAccessTokenHolder.java
package cc.coopersoft.accesstoken.api;

import cc.coopersoft.accesstoken.api.callback.CodeRequestCallback;
import cc.coopersoft.accesstoken.api.callback.VerificationCallback;
import cc.coopersoft.accesstoken.interceptor.AccessTokenInterceptor;

public interface PhoneSupportAccessTokenHolder extends AccessTokenHolder{

  void sendAuthenticationCode(String phoneNumber, CodeRequestCallback callback);

  void sendVerificationCode(String phoneNumber, CodeRequestCallback callback);

  void sendRegistrationCode(String phoneNumber, CodeRequestCallback callback);

  void verificationCode(String phoneNumber, String code, VerificationCallback callback);

}

File: keycloak-client/src/main/java/cc/coopersoft/accesstoken/api/PhoneSupportAccessTokenHolder.kt
package cc.coopersoft.accesstoken.api

import cc.coopersoft.accesstoken.api.callback.CodeRequestCallback
import cc.coopersoft.accesstoken.api.callback.VerificationCallback

interface PhoneSupportAccessTokenHolder : AccessTokenHolder {
    fun sendAuthenticationCode(phoneNumber: String?, callback: CodeRequestCallback?)
    fun sendVerificationCode(phoneNumber: String?, callback: CodeRequestCallback?)
    fun sendRegistrationCode(phoneNumber: String?, callback: CodeRequestCallback?)
    fun verificationCode(phoneNumber: String?, code: String?, callback: VerificationCallback?)
}